src/lib.rs

```diff
@@ -376,6 +551,99 @@ impl Contract {
             .create_community_account(new_community.handle.clone())
     }
 
+    #[payable]
+    pub fn edit_proposal(
+        &mut self,
+        id: ProposalId,
+        body: VersionedProposalBody,
+        labels: HashSet<String>,
+    ) {
+        near_sdk::log!("edit_proposal");
+        require!(
+            self.is_allowed_to_edit_proposal(id, Option::None),
+            "The account is not allowed to edit this proposal"
+        );
+        let editor_id = env::predecessor_account_id();
+        let mut proposal: Proposal =
+            self.proposals.get(id.into()).unwrap_or_else(|| panic!("Proposal id {} not found", id)).into();
```

Comment:
Are these intos necessary?

```suggestion
            self.proposals.get(id).unwrap_or_else(|| panic!("Proposal id {} not found", id));
```

Reply:
Yes, the for first one an `id` is an index of a vector. And it should be u64. But ProposalId is u32. So I would change it for id.into()
For the second one, it is casting from `VersionedProposal` to `Proposal`